### ISSUE TITLE
Is host memory used during cudf.serialize?

### ISSUE THREAD

**Issue by mjshare** [rapidsai/cudf#12348]
cudf description

class Serializable:
    """A serializable object composed of device memory buffers.

    This base class defines a standard serialization protocol for objects
    encapsulating device memory buffers. Serialization proceeds by copying
    device data onto the host, then returning it along with suitable metadata
    for reconstruction of the object. Deserialization performs the reverse
    process, copying the serialized data from the host to new device buffers.
    Subclasses must define the abstract methods :meth:`~.serialize` and
    :meth:`~.deserialize`. The former defines the conversion of the object
    into a representative collection of metadata and data buffers, while the
    latter converts back from that representation into an equivalent object.
    """

**Comment by wence-** [comment]
tl;dr: no, unless you are using it through the pickle interface.

If you call `serialize()` on a cudf object, you won't get host memory being used (this is used in dask/distributed to ensure that in a multi-GPU setting we do direct GPU-to-GPU transfer of the data). The metadata "header" that describes the object that has been serialized does use host memory (it's just a normal python dictionary).

Through the serialization interface, cudf also supports the Python's pickle protocol (so you can do `pickle.dump` and `pickle.load` with cudf objects). In "out of band" protocol5 mode, this doesn't serialize the device buffers "in-band" but it does need to bring them to host, because we have to deliver `memoryview` objects which are wrappers around host pointers.

**Comment by vyasr** [comment]
I'm going to close this as answered, feel free to reopen if there are further questions though.
